onnxruntime/test/providers/cpu/ml/label_encoder_test.cc

```diff
@@ -42,5 +42,131 @@ TEST(LabelEncoder, IntToString) {
+
+  const std::vector<std::int64_t> keys{9, 1, 4};
+  const std::vector<std::string> values{"AA", "BB", "DD"};
+
+  test.AddAttribute("keys_int64s", keys);
+  test.AddAttribute("values_strings", values);
+  test.AddAttribute<std::string>("default_string", "CC");
+
+  test.AddInput<std::int64_t>("X", dims, input);
+  test.AddOutput<std::string>("Y", dims, output);
+
+  test.Run();
+}
+
+TEST(LabelEncoder, FloatToStringOpset2) {
+  std::vector<std::int64_t> dims{5, 1};
+
+  std::vector<float> input{9.4f, 1.7f, 3.6f, 1.2f, 2.8f};
+  std::vector<std::string> output{"AA", "BB", "DD", "CC", "CC"};
+
+  OpTester test("LabelEncoder", 2, onnxruntime::kMLDomain);
+
+  const std::vector<float> keys{9.4f, 1.7f, 3.6f};
+  const std::vector<std::string> values{"AA", "BB", "DD"};
+
+  test.AddAttribute("keys_floats", keys);
+  test.AddAttribute("values_strings", values);
+  test.AddAttribute<std::string>("default_string", "CC");
+
+  test.AddInput<float>("X", dims, input);
+  test.AddOutput<std::string>("Y", dims, output);
+
+  test.Run();
+}
+
+TEST(LabelEncoder, StringToFloatOpset2) {
+  std::vector<std::int64_t> dims{5, 1};
+
+  std::vector<std::string> input{"AA", "BB", "DD", "CC", "CC"};
+  std::vector<float> output{9.4f, 1.7f, 3.6f, 55.66f, 55.66f};
+
+  OpTester test("LabelEncoder", 2, onnxruntime::kMLDomain);
+
+  const std::vector<std::string> keys{"AA", "BB", "DD"};
+  const std::vector<float> values{9.4f, 1.7f, 3.6f};
+
+  test.AddAttribute("keys_strings", keys);
+  test.AddAttribute("values_floats", values);
+  test.AddAttribute("default_float", 55.66f);
+
+  test.AddInput<std::string>("X", dims, input);
+  test.AddOutput<float>("Y", dims, output);
+
+  test.Run();
+}
+
+TEST(LabelEncoder, FloatToInt64Opset2) {
+  std::vector<std::int64_t> dims{5};
+
+  std::vector<float> input{9.4f, 1.7f, 3.6f, 55.66f, 55.66f};
+  std::vector<std::int64_t> output{1, 9, 3, -8, -8};
+
+  OpTester test("LabelEncoder", 2, onnxruntime::kMLDomain);
+
+  const std::vector<float> keys{9.4f, 1.7f, 3.6f};
+  const std::vector<std::int64_t> values{1, 9, 3};
+
+  test.AddAttribute("keys_floats", keys);
+  test.AddAttribute("values_int64s", values);
+  test.AddAttribute("default_int64", (std::int64_t)-8);
+
+  test.AddInput<float>("X", dims, input);
+  test.AddOutput<std::int64_t>("Y", dims, output);
+
+  test.Run();
+}
+
+TEST(LabelEncoder, Int64ToFloatOpset2) {
+  std::vector<std::int64_t> dims{5};
+
+  std::vector<std::int64_t> input{3, 1, 9, -8, -8};
+  std::vector<float> output{3.6f, 9.4f, 1.7f, 55.66f, 55.66f};
+
+  OpTester test("LabelEncoder", 2, onnxruntime::kMLDomain);
+
+  const std::vector<std::int64_t> keys{1, 9, 3};
+  const std::vector<float> values{9.4f, 1.7f, 3.6f};
+
+  test.AddAttribute("keys_int64s", keys);
+  test.AddAttribute("values_floats", values);
+  test.AddAttribute("default_float", 55.66f);
+
+  test.AddInput<std::int64_t>("X", dims, input);
+  test.AddOutput<float>("Y", dims, output);
+
+  test.Run();
+}
+
```

Comment:
Do we need a test to catch some failing conditions-- i.e. if both **keys_strings** and **keys_floats** attributes are set? Users are likely to configure an operator instance with this type of mistake.

Reply:
It has been checked in ONNX's shape inference function.